modules/apps/27-interchain-accounts/controller/keeper/handshake.go

```diff
@@ -70,7 +70,12 @@ func (k Keeper) OnChanOpenInit(
 			return "", sdkerrors.Wrapf(icatypes.ErrActiveChannelAlreadySet, "existing active channel %s for portID %s is already OPEN", activeChannelID, portID)
 		}
 
-		if !icatypes.IsPreviousMetadataEqual(channel.Version, metadata) {
+		appVersion, found := k.GetAppVersion(ctx, portID, activeChannelID)
+		if !found {
+			panic(fmt.Sprintf("active channel mapping set for %s, but channel does not exist in channel store", activeChannelID))
```

Comment:
a nit: `channel does not exist in channel store` is kind of confusing if the error is `GetAppVersion` returning not found -- maybe something like channel version does not exist?